tuning/acceleration/plugins/framework_plugin_unsloth_autogptq.py

```diff
@@ -0,0 +1,150 @@
+
+def parse_configuration():
+    pass
+
+import torch
+from transformers import TrainingArguments
+from peft import LoraConfig
+from peft.tuners.lora.model import LoraModel
+from auto_gptq.utils.peft_utils import GPTQLoraModel
+from unsloth import FastLanguageModel
+from unsloth.utils.modeling import QuantizationMethod
+from unsloth.gptq.triton.layers import GPTQuantLinear
+
+from types import MethodType
+from typing import Tuple, List, Dict
+
+from .framework_plugin import AccelerationPlugin
+from .framework_plugin_autogptq import _replace_module, LoraLinearGPTQ
+
+# def _dispatch_lora_linear_for_gptq(
+def _create_new_module_triton(
+    lora_config: LoraConfig,
+    adapter_name: str,
+    target: torch.nn.Module,
+    **kwargs,
+):
+    # if the base layer module matches a supported class, dispatch the lora linear
+    # to be installed
+    new_module = target
+    if (
+        isinstance(target, GPTQuantLinear)
+    ):
+        new_module = LoraLinearGPTQ(target, adapter_name, **kwargs)
+    # if module cannot be found, return None which results in a raise in the call-stack
+    return new_module
+
+class UnslothAutoGPTQAccelerationPlugin(AccelerationPlugin):
+    
+    require_packages = ['auto_gptq', 'unsloth']
```

Comment:
need to add a required model archs here since we are limited to only few models (e.g. llama, mistral, mixtarl)